code/modules/mob/living/carbon/hallucination.dm

```diff
@@ -0,0 +1,1102 @@
+var/list/hallucinated_phrases = file2list("config/hallucination/hallucinated_phrases.txt")
+var/list/hallucinated_actions = file2list("config/hallucination/hallucinated_actions.txt")	//important note when adding to this file: "you" will always be replaced by the hallucinator's name
+var/list/hallucinated_thoughts = file2list("config/hallucination/hallucinated_thoughts.txt")
+
+
+/mob/living/carbon/var/next_hallucination = 0			//Hallucination spam limit var
+/mob/living/carbon/var/list/hallucinations = list()		//Hallucinations currently affecting the mob. Not to be confused with singular "hallucination" which is a NUM variable like confused/drowsy/eye_blind etc
+
+///////////////////////////////////////Hallucinated Hearing///////////////////////////
+/mob/living/carbon/hear_say(var/message, var/verb = "says", var/datum/language/language = null, var/alt_name = "",var/italics = 0, var/mob/speaker = null, var/sound/speech_sound, var/sound_vol)
+	if(hallucination >= 60 && prob(1))
+		var/orig_message = message
+		message = LAZYPICK(hallucinated_phrases, message)
+		log_say("Hallucination level changed [orig_message] by [speaker] to [message] for [key_name(src)].", ckey=key_name(src))
+	..()
+/mob/living/carbon/hear_radio(var/message, var/verb="says", var/datum/language/language=null, var/part_a, var/part_b, var/mob/speaker = null, var/hard_to_hear = 0, var/vname ="")
+	if(hallucination >= 60 && prob(1))
+		var/orig_message = message
+		message = LAZYPICK(hallucinated_phrases, message)
+		log_say("Hallucination level changed [orig_message] by [speaker] to [message] for [key_name(src)].", ckey=key_name(src))
+	..()
+
+/mob/living/carbon/proc/handle_hallucinations()     //Main handling proc, called in life()
+
+	hallucination -= 1	//Tick down the duration
+
+	if(!hallucination)  //We're done
+		return
+
+	if(!client || stat || world.time < next_hallucination)
+		return
+
+	var/hall_delay = rand(160,250)		//Time between hallucinations, modified by switch below.
+
+	switch(hallucination)	//26-74 are intentionally left off, as they do not modify the delay. This is a pretty common range for hallucinations.
+		if(1 to 25)		//Winding down, less frequent.
+			hall_delay *= 2
+		if(75 to 399)		//Yo mind really fucked, more frequent.
+			hall_delay *= 0.75
+		if(400 to INFINITY)		//This should only be possible in cult conversions. Very low delay to represent your flayed mind.
+			hall_delay *= 0.25
+
+	next_hallucination = world.time + hall_delay
+	var/list/candidates = list()
+	for(var/T in subtypesof(/datum/hallucination/))
+		var/datum/hallucination/H = new T
+		if(H.can_affect(src))
+			candidates += H
+	if(candidates.len)
+		var/datum/hallucination/H = pick(candidates)
+		H.holder = src
+		H.activate()
```

Comment:
Yeah, no, don't do this every life tick. Make a new subsystem or use the brain trauma one to initialize a list of hallucinations at roundstart and then pick one from the list.

Reply:
It won't do this candidate check on every life tick. It will only do it if the `if(!client || stat || world.time < next_hallucination)` check passes.

Reply:
Additionally, it's impossible to just initialize a list at roundstart if we want these to be varied, since the can_affect() check is what checks for if there are mobs around, how badly they're hallucinating, if they're deaf, etc. which will give us candidates that can currently affect the person. However I will shove this checking and all of the lists into a subsystem since that's a way better idea.